File: linkedlist/reversell.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

def reverse_linked_list(head):
    prev_node = None
    current_node = head
    while current_node:
        next_node = current_node.next
        current_node.next = prev_node
        prev_node = current_node
        current_node = next_node
    return prev_node

def print_linked_list(head):
    current_node = head
    while current_node:
        print(current_node.data, end=" ")
        current_node = current_node.next
    print()

# Function to create linked list from user input
def create_linked_list_from_input():
    nums = input("Enter space-separated numbers for linked list: ").split()
    head = None
    for num in nums:
        if not head:
            head = Node(int(num))
            tail = head
        else:
            tail.next = Node(int(num))
            tail = tail.next
    return head

if __name__ == "__main__":
    head = create_linked_list_from_input()

    print("Original linked list:")
    print_linked_list(head)

    head = reverse_linked_list(head)

    print("Reversed linked list:")
    print_linked_list(head)

File: linkedlist/reversell.cpp
#include <iostream>
using namespace std;

struct Node {
	int data;
	struct Node* next;
};

// Function to reverse the linked list
void reverse(Node** head)
{
    Node* current = *head;
    Node* prev = NULL;
    Node* next = NULL;
    while (current != NULL) {
        next = current->next;
        current->next = prev;
        prev = current;
        current = next;
    }
    *head = prev;
}

// Utility function to create a new node
Node* newNode(int key)
{
	Node* temp = new Node;
	temp->data = key;
	temp->next = NULL;
	return temp;
}

// Function to create a linked list from user input
Node* createLinkedListFromInput()
{
    cout << "Enter the numbers for the linked list (enter -1 to stop):" << endl;
    Node* head = nullptr;
    Node* tail = nullptr;
    int num;
    while (true) {
        cin >> num;
        if (num == -1)
            break;
        if (!head) {
            head = newNode(num);
            tail = head;
        } else {
            tail->next = newNode(num);
            tail = tail->next;
        }
    }
    return head;
}

// Utility function to print a linked list
void printlist(Node* head)
{
	while (head != NULL) {
		cout << head->data << " ";
		head = head->next;
	}
	cout << endl;
}

// Driver code
int main()
{
	Node* head1 = createLinkedListFromInput();
	cout << "Given linked list\n";
	printlist(head1);
	reverse(&head1);
	cout << "Reversed linked list\n";
	printlist(head1);
	return 0;
}
